internal/gather/gather.go

```diff
@@ -0,0 +1,270 @@
+/*
+SPDX-License-Identifier: Apache-2.0
+
+Copyright Contributors to the Submariner project.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gather
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/submariner-io/admiral/pkg/reporter"
+	"github.com/submariner-io/submariner-operator/internal/cli"
+	"github.com/submariner-io/submariner-operator/internal/component"
+	"github.com/submariner-io/submariner-operator/internal/constants"
+	"github.com/submariner-io/submariner-operator/internal/exit"
+	"github.com/submariner-io/submariner-operator/internal/restconfig"
+	"github.com/submariner-io/submariner-operator/pkg/brokercr"
+	"github.com/submariner-io/submariner-operator/pkg/client"
+	"github.com/submariner-io/submariner-operator/pkg/cluster"
+	"github.com/submariner-io/submariner-operator/pkg/names"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+)
+
+type Options struct {
+	Directory            string
+	IncludeSensitiveData bool
+	ModuleFlags          map[string]bool
+	TypeFlags            map[string]bool
+}
+
+const (
+	Logs      = "logs"
+	Resources = "resources"
+)
+
+var ModuleFlags = map[string]bool{
+	component.Connectivity:     false,
+	component.ServiceDiscovery: false,
+	component.Broker:           false,
+	component.Operator:         false,
+}
```

Comment:
This shouldn’t be necessary, the same information is effectively present in `gatherFuncs`; perhaps a `GetSupportedModules() []string` function could be used instead, or an `IsModuleSupported(module string) bool` function.